linter/lintmain/internal/check/check.go

```diff
@@ -422,33 +422,62 @@ func loadPackages(cfg *packages.Config, patterns []string) ([]*packages.Package,
 		return nil, err
 	}
 
-	result := pkgs[:0]
-
-	// First pkgs traversal selects external tests and
-	// packages built for testing.
-	// If there is no tests for the package,
-	// we're going to check them during the second traversal
-	// which visits normal package if only it was
-	// not checked during the first traversal.
-	withTests := make(map[string]bool)
-	for _, pkg := range pkgs {
-		if !strings.Contains(pkg.ID, ".test]") {
-			continue
+	type pack struct {
+		base         *packages.Package
+		internalTest *packages.Package
+		externalTest *packages.Package
+	}
+	packs := make(map[string]*pack)
+	internPack := func(key string) *pack {
+		if p := packs[key]; p != nil {
+			return p
+		}
+		p := &pack{}
+		packs[key] = p
+		return p
+	}
+	mustBeNil := func(pkg *packages.Package) {
+		if pkg != nil {
+			panic(fmt.Sprintf("nil assertion failed"))
```

Comment:
`fmt.Sprintf` is not needed here 😉 
or `pkg.Foo` as a param is missing

Reply:
Ooo-ops. Thanks for pointing that out.